From 7850481bdb4ca4314def1661bf61dc09921f60a7 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:51 -0400
Subject: [net] openvswitch: Fix output of SCTP mask

Message-id: <0fa51f15b39cfbf4551f5d14dbac3f585dbe263e.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94033
O-Subject: [RHEL7.1 net PATCH 18/50] openvswitch: Fix output of SCTP mask.
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit d92ab13558599cf73bbc269ce257fe16575d327a
Author: Jarno Rajahalme <jrajahalme@nicira.com>
Date:   Thu Mar 27 12:47:11 2014 -0700

    openvswitch: Fix output of SCTP mask.

    The 'output' argument of the ovs_nla_put_flow() is the one from which
    the bits are written to the netlink attributes.  For SCTP we
    accidentally used the bits from the 'swkey' instead.  This caused the
    mask attributes to include the bits from the actual flow key instead
    of the mask.

    Signed-off-by: Jarno Rajahalme <jrajahalme@nicira.com>
    Acked-by: Pravin B Shelar <pshelar@nicira.com>
    Signed-off-by: Jesse Gross <jesse@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/flow_netlink.c b/net/openvswitch/flow_netlink.c
index 84caa99..32a725c 100644
--- a/net/openvswitch/flow_netlink.c
+++ b/net/openvswitch/flow_netlink.c
@@ -1059,11 +1059,11 @@ int ovs_nla_put_flow(const struct sw_flow_key *swkey,
     goto nla_put_failure;
    sctp_key = nla_data(nla);
    if (swkey->eth.type == htons(ETH_P_IP)) {
-    sctp_key->sctp_src = swkey->ipv4.tp.src;
-    sctp_key->sctp_dst = swkey->ipv4.tp.dst;
+    sctp_key->sctp_src = output->ipv4.tp.src;
+    sctp_key->sctp_dst = output->ipv4.tp.dst;
    } else if (swkey->eth.type == htons(ETH_P_IPV6)) {
-    sctp_key->sctp_src = swkey->ipv6.tp.src;
-    sctp_key->sctp_dst = swkey->ipv6.tp.dst;
+    sctp_key->sctp_src = output->ipv6.tp.src;
+    sctp_key->sctp_dst = output->ipv6.tp.dst;
    }
   } else if (swkey->eth.type == htons(ETH_P_IP) &&
       swkey->ip.proto == IPPROTO_ICMP) {
-- 
1.7.1